File: python/src/LeftAndRightSumDifferences2574.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int numsSum = Arrays.stream(nums).sum();
        int leftSum = 0;
        int numsLen = nums.length;
        int[] result = new int[numsLen];
        for(int i = 0; i < numsLen; i++) {
            numsSum -= nums[i];
            result[i] = Math.abs(numsSum - leftSum);
            leftSum += nums[i];
        }
        return result;
    }
}


class LeftAndRightSumDifferences2574 {
    public static void main(String[] args) throws Exception {
        Solution solution = new Solution();
        int[] nums = { 1, 2, 3, 4, 5 };
        int[] result = solution.leftRightDifference(nums);
        if (result[0] != 14 || result[1] != 10 || result[2] != 6 || result[3] != 2 || result[4] != 0) {
            throw new Exception("Invalid left and right sum differences");
        }
        nums = new int[] { 1 };
        result = solution.leftRightDifference(nums);
        if (result[0] != 0) {
            throw new Exception("Invalid left and right sum differences");
        }
        nums = new int[] { 1, 2 };
        result = solution.leftRightDifference(nums);
        if (result[0] != 1 || result[1] != 1) {
            throw new Exception("Invalid left and right sum differences");
        }
        nums = new int[] { 1, 2, 3 };
        result = solution.leftRightDifference(nums);
        if (result[0] != 4 || result[1] != 2 || result[2] != 0) {
            throw new Exception("Invalid left and right sum differences");
        }
        nums = new int[] { 1, 2, 3, 4 };
        result = solution.leftRightDifference(nums);
    }
}

File: python/src/left_and_right_sum_differences_2574.py
from typing import List


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)
        numsSum = sum(nums)
        leftSum = 0
        for i in range(len(nums)):
            numsSum -= nums[i]
            result[i] = abs(numsSum - leftSum)
            leftSum += nums[i]
        return result


def test():
    testcases = [
        [[10,4,8,3], [15,1,11,22]],
        [[1], [0]],
        [[1, 2], [1, 1]],
        [[1, 2, 3], [2, 0, 2]],
        [[1, 2, 3, 4], [6, 4, 2, 0]],
        [[1, 2, 3, 4, 5], [10, 8, 6, 4, 0]],
        [[1, -1], [0, 0]],
        [[-1, -2], [0, 0]],
        [[-1, -2, -3], [0, 0, 0]],
        [[-1], [0]],
        [[-1, -2], [1, 1]],
        [[-1, -2, -3], [2, 0, 2]],
        [[-1, -2, -3, -4], [6, 4, 2, 0]],
        [[-1, -2, -3, -4, -5], [10, 8, 6, 4, 0]],
        [[-1, -2], [1, 1]],
        [[-1], [0]],
        [[-1, -2], [1, 1]],
        [[-1, -2], [1, 1]],
    ]
    solution = Solution()
    for index, testcase in enumerate(testcases):
        result = solution.leftRightDifference(testcase[0])
        assert result == testcase[1], "Testcase #{} failed! Expected: {}, Got: {}".format(
            index, testcase[1], result)


if __name__ == "__main__":
    test()
